### DISCUSSION TITLE
Dynamic query navigation is broken

### DISCUSSION THREAD

**Discussion by noobCode-69** [vercel/next.js#63456] [Help] [answered]
### Summary

Hey @icyJoseph 

I have a route which is highly controlled by dynamic queries ex "?a=something&b=something-else" . I have a button (search button) on my page which when clicked navigated to the same route with may different or same query . My issue is

Suppose I'm at "?a=a&b=b" and i navigate to "?a=b&b=b" again then my routing is never complete ( I have a progress bar on top which never gets completed ) , on further investigation i found that when navigating to the same queries my app doesn't make a trip to the nextjs server.

I'm doing something wrong

**Comment by icyJoseph** [comment]
How is the progress bar implemented?

**Reply by noobCode-69** [reply]
Hey , I'm using the [next-progress-bar](https://www.npmjs.com/package/next-nprogress-bar). 
My  Issue is that my web app is not making a trip to the server , when i do router.push() with the same params as my current route .
But I want that behaviour.

**Comment by noobCode-69** [comment]
Hey @icyJoseph  here is the min reproduction [repo](https://github.com/noobCode-69/min-repoduction) and here is the deployed [link](https://min-repoduction.vercel.app/?c) . Here is my intented behaviour 
1) When I click "a" and press navigate , I get a new server generated number (Good)
2) Then I click "b" and press navigate, I get a new server generated number (Good)
3) Then I click "b" and press navigate, I don't get a new server generated number (Bad)

**Reply by icyJoseph** [reply, marked as answer]
So the issue you have is that when you navigate to the exact same route, then you need to `router.refresh()` to invalidate the cache:

```tsx
 const navigate = () => {
    router.push(`/?${s}`);
    router.refresh();
  };
```

Why we have to do this is explained here: 

- https://nextjs.org/docs/app/building-your-application/caching#4-nextjs-caching-on-the-client-router-cache
- https://nextjs.org/docs/app/building-your-application/caching#5-subsequent-navigations
